framework/src/auxkernels/VectorPostprocessorVisualizationAux.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "VectorPostprocessorVisualizationAux.h"
+
+registerMooseObject("MooseApp", VectorPostprocessorVisualizationAux);
+
+template <>
+InputParameters
+validParams<VectorPostprocessorVisualizationAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+
+  params.addClassDescription("Read values from a VectorPostprocessor that is producing vectors "
+                             "that are 'number of processors' * in length.  Puts the value for "
+                             "each processor into an elemental auxiliary field.");
+
+  params.addRequiredParam<VectorPostprocessorName>(
+      "vpp", "The name of the VectorPostprocessor to pull the data from.");
+  params.addRequiredParam<std::string>(
+      "vector_name", "The name of the vector to use from the VectorPostprocessor");
+
+  return params;
+}
+
+VectorPostprocessorVisualizationAux::VectorPostprocessorVisualizationAux(
+    const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _vpp_vector(getVectorPostprocessorValue("vpp", getParam<std::string>("vector_name"))),
```

Comment:
Interesting that this works here. It actually isn't correct. We have all of those "ByName" variants, which take the parameter name (instead of the actual name). This might mean we have another bug in the retrieval of VPPs.

Reply:
I don't think this is a bug. The not-byName variants fall back to byName if no parameter by that name exists. This is for backwards compatibility, but definitely should not be used in new code.

Reply:
Never mind. The byName issue affects only the first parameter

Reply:
It's right the way it is - it is using the _non_ "by name" version on purpose.  `vpp` is the parameter name.

There is no such thing for the `vector_name` though (which is why I have to do `getParam()` for it).  It could be argued that we should make that part of the interface too...

Reply:
👍